proto/internal/temporal/server/api/historyservice/v1/request_response.proto

```diff
@@ -638,6 +638,75 @@ message UpdateWorkflowExecutionResponse {
     temporal.api.workflowservice.v1.UpdateWorkflowExecutionResponse response = 1;
 }
 
+message ReadHistoryBranchRequest {
+    string namespace_id = 1;
+    int32 shard_id = 2;
+    bytes branch_token = 3;
+    int64 min_event_id = 4;
+    int64 max_event_id = 5;
+    int64 page_size = 6;
+    bytes next_page_token = 7;
+}
+
+message ReadHistoryBranchResponse {
+    repeated temporal.api.history.v1.HistoryEvent history_events = 1;
+    bytes next_page_token = 2;
+    int64 size = 3;
+}
+
+message ReadHistoryBranchReverseRequest {
+    string namespace_id = 1;
+    int32 shard_id = 2;
+    bytes branch_token = 3;
+    int64 max_event_id = 4;
+    int64 page_size = 5;
+    int64 last_first_transaction_id = 6;
+    bytes next_page_token = 7;
+}
```

Comment:
It's better to allow clients to provide an `order_by` field in the original request rather than defining a second endpoint: https://google.aip.dev/132#ordering

Reply:
I originally was going to go that route as well, except if you look at the fields in the request object for the forward vs reverse, you'll notice that they're not just opposite ordering of each other.

Reply:
We discussed this offline, so I'll recap here: I wanted us to follow AIP guidelines because that would give us a much more standard API. However, we have a strong convention for our API names and arguments to map 1:1 with the signatures of the objects that implement them. This argument didn't sway me initially, but I now think it's worth it to keep this practice since it's so ingrained in the server codebase.